### ISSUE TITLE
Using ?: conditional operator fails

### ISSUE THREAD

**Issue by mehrt** [aspnet/EntityFramework#3585]
This may be the same issue as #649, but I'm not sure. If it is, I'm sorry to have wasted your time and bandwidth.

I'm using beta 8. Model:
```
public class Item
{
	public int Id { get; set; }
	public string First { get; set; }
	public string Second{ get; set; }
}

public class MyContext : DbContext
{
	public DbSet<Item> Items{ get; set; }

	protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
	{
		optionsBuilder.UseSqlServer(@"Server=SomeServer;Database=SomeDb;User Id=SomeUser;Password=SomePW");
	}
}
```
If I then try ...
```
using (var ctx = new MyContext())
{
	ctx.Database.EnsureCreated();

	string first = "12345", second = "";
	var r = ctx.Items.Where(
		i => i.First == first && 
			(second.Length == 5 ? i.Second == second
								: (i.Second == string.Empty || i.Second == null)))
	.FirstOrDefault();
}
```
... I get a similar error as described in #717:
*An expression of non-boolean type specified in a context where a condition is expected, near 'THEN'.*
If I try this instead ...

```
var r = ctx.Items.Where(
    i => i.First == first &&
         (second.Length == 5 && i.Second == second || i.Second == string.Empty || i.Second == null))
    .FirstOrDefault();
```
... I get the expected result.

**Comment by mikary** [comment]
Attempted to reproduce this on current dev and it appears the problem was resolved in 10f74f9d.

No exception is thrown and the following SQL is generated for the first query:
```sql
SELECT TOP(1) [i].[Id], [i].[First], [i].[Second]
FROM [Items] AS [i]
WHERE ([i].[First] = @__first_0) AND (((@__second_Length_1 = 5) AND ([i].[Second] = @__second_2)) OR ((@__second_Length_1 <> 5) AND (([i].[Second] = @__Empty_3) OR [i].[Second] IS NULL)))
```

**Comment by mikary** [comment]
Note: Regression testing for this issue should be covered by `QueryTestBase.Where_ternary_boolean_condition_with_another_condition` added in commit 10f74f9db2
